### PR TITLE
feat(server): add ModernRsbuildServer

### PR DESCRIPTION
**Pull request by 9aoy**
## Summary

The old modern server should be replaced by ModernRsbuildServer when rsbuild is fully integrated into modern.js.

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/server/server/src/server/devServer.ts [hunk 1]
```diff
@@ -84,48 +115,65 @@ export class ModernDevServer extends ModernServer {
...
-    const { befores, afters } = this.applySetupMiddlewares();
+    this.rsbuild.onDevCompileDone(({ stats }) => {
+      // Reset only when client compile done
+      if (stats.toJson({ all: false }).name !== 'server') {
+        this.onRepack({ routes: this.getRoutes() });
+      }
+    });
 
     // before dev handler
     const beforeHandlers = await this.setupBeforeDevMiddleware();
```
Comment (zllkjc) [inline packages/server/server/src/server/devServer.ts:164]: It seems that the order here is inconsistent.
Origin: before - > hookBefore - > unshift - > push - > hookAfter - > after
Now: hookBefore - > before - > unshift - > push - > after - > hookAfter

However, these two hooks are not used. we can remove them, and developer won't need to use them later.

File: packages/server/server/src/server/devServerOld.ts [hunk 2]
```diff
@@ -0,0 +1,462 @@
+import { createServer, IncomingMessage, Server, ServerResponse } from 'http';
```
Comment (zllkjc) [inline packages/server/server/src/server/devServerOld.ts:1]: remove this file ? and also remove the dependencies like "http-compression"

File: packages/server/server/src/server/devServer.ts (hunk 1 above)
Comment (9aoy) [reply packages/server/server/src/server/devServer.ts:164]: emm... I will remove these two hooks  in next PR

File: packages/server/server/src/server/devServerOld.ts (hunk 2 above)
Comment (9aoy) [reply packages/server/server/src/server/devServerOld.ts:1]: The old modern server and deps should be replaced by ModernRsbuildServer after rsbuild is fully integrated into modern.js. Now, we still need it.

File: packages/server/server/src/server/devServer.ts (hunk 1 above)
Comment (9aoy) [reply packages/server/server/src/server/devServer.ts]: updated!

**Review by zllkjc — APPROVED** [review]

**Pull request merged**